src/AnimationLC.cpp

```diff
@@ -11,24 +11,35 @@
 
 void AnimationLC::frameRendered(const Ogre::FrameEvent& evt) {
     for (auto anim : animations)
-        if (anim.second->getEnabled())
-            anim.second->addTime(evt.timeSinceLastFrame);
+        if (anim.second->getEnabled()) {
+            if (anim.second->hasEnded()) {
+                anim.second->setEnabled(false);
+            } else {
+                anim.second->addTime(evt.timeSinceLastFrame);
+            }
+        }
 }
 
 void AnimationLC::startAnimation(std::string name) {
+    animations.find(name)->second->setTimePosition(0.0);
     animations.find(name)->second->setEnabled(true);
```

Comment:
Consider changing this to the following:

```cpp
auto animation = animations.find(name)->second;
animation->setTimePosition(0.0);
animation->setEnabled(true);
```

Repeated look-ups can decrease performance.